### PR TITLE
Add MPI SFT Benchmark for Qwen 2.5 on GSM8K dataset

### PR DESCRIPTION
**Pull request by kapil27**
This commit introduces a new multi-node MPI performance benchmark for training the Qwen 2.5 7B model using Supervised Fine-Tuning (SFT) on the GSM8K dataset. The benchmark includes a README file detailing the setup, prerequisites, and scaling strategies, along with a training script (`train_sft.py`) that utilizes PyTorch's Distributed Data Parallel (DDP) with an MPI backend. Additionally, a Kubernetes `trainjob.yaml` configuration is provided for deploying the benchmark in a Kubeflow environment.

## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

## Summary by CodeRabbit

* **Documentation**
  * Added a README describing setup, deployment, scaling, configurable env vars, expected output/metrics, and cleanup for a new MPI-based SFT training workflow.

* **New Features**
  * Added an MPI-distributed SFT training entrypoint that runs PyTorch DDP with MPI, captures per-step timing and GPU memory metrics, prints consolidated benchmark results, and a Kubernetes TrainJob manifest to deploy and scale the workload with resource and affinity settings.

### CODE REVIEW

File: benchmarks/kftv2-mpi-sft/train_sft.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+        "dataset_batch_size": 1000,
+        "packing": False,
+        "max_steps": int(os.environ.get("SFT_MAX_STEPS", "50")),
+        "per_device_train_batch_size": int(os.environ.get("SFT_BATCH_SIZE", "1")),
+        "per_device_eval_batch_size": 1,
+        "auto_find_batch_size": False,
+        "eval_strategy": "no", # Disable eval for benchmark
+        "bf16": True,
+        "tf32": False,
+        "learning_rate": float(os.environ.get("SFT_LEARNING_RATE", "5e-7")),
```
[the remark this answers is not in the document]
Comment (kapil27) [reply benchmarks/kftv2-mpi-sft/train_sft.py]: Thanks for the review! The `SFT_DATASET_SIZE` is specifically used to truncate the dataset (e.g., for faster testing), but it is not intended to dynamically alter the learning rate. The learning rate should remain at the default `5e-7` regardless of whether the dataset is truncated or not, so keeping them independent is the intended behavior.
